### DISCUSSION TITLE
Checkout submodules with branch

### DISCUSSION THREAD

**Discussion by aspirisen** [actions/checkout#839] [Q&A]
Hello, looks like github actions does not respect "branch" property in `.gitmodules` files.
When I clone the repo locally the appropriate branches are available, but on CI the i have detached head

**Comment by henrygriffiths** [comment]
Hi @aspirisen,

Are you setting the `submodules` property to either `true` or `recursive` when using the action?

**Comment by aspirisen** [comment]
@henrygriffiths hi, i have `submodules: true`

**Reply by henrygriffiths** [reply]
Hi @aspirisen,

Have you tried setting `submodules` to `recursive` instead? Do you have a repo available to look at?
